### PR TITLE
update pre commit config for terraform_fmt & added app_gateway_backend_check 

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
update pre commit config for terraform_fmt & added app_gateway_backend_check 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,4 +1,9 @@
...
-    - id: terraform_fmt
+repos:
+- repo: https://github.com/antonbabenko/pre-commit-terraform
+  rev: v1.74.1
+  hooks:
+    - id: terraform_fmt
+- repo: https://github.com/hmcts/pre-commit-hooks
+  rev: 0.0.2
+  hooks:
+    - id: app_gateway_backend_check
```
Comment (timja) [inline .pre-commit-config.yaml:9]: could you add some documentation on this please?

Both to the readme here:
https://github.com/hmcts/pre-commit-hooks

and probably also to the README in this repo and maybe a CONTRIBUTING.md file?

Comment (thomast1906) [reply .pre-commit-config.yaml:9]: Small readme added to this PR

File: README.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
-# common-platform-terraform
-All things terraform
-
+# common-platform-terraform
+All things terraform
+
+## Pre-commit Installation
+This repository has been configured with pre-commit hooks. To install pre-commit hook, see [here](https://github.com/hmcts/pre-commit-hooks#pre-commit-installation)
```
Comment (timja) [inline README.md:5]:
```suggestion
This repository has been configured with pre-commit hooks. To install pre-commit see its [installation guide](https://pre-commit.com/#installation)

```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
-# common-platform-terraform
-All things terraform
-
+# common-platform-terraform
+All things terraform
+
+## Pre-commit Installation
+This repository has been configured with pre-commit hooks. To install pre-commit hook, see [here](https://github.com/hmcts/pre-commit-hooks#pre-commit-installation)
+
+Additional for this repository, additional dependencies:
```
Comment (timja) [inline README.md:7]:
```suggestion
For this repository you will also need these dependencies: 

```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
...
-
+# common-platform-terraform
+All things terraform
+
+## Pre-commit Installation
+This repository has been configured with pre-commit hooks. To install pre-commit hook, see [here](https://github.com/hmcts/pre-commit-hooks#pre-commit-installation)
+
+Additional for this repository, additional dependencies: 
+- terraform [Installation here](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+- yq [Installation here](https://github.com/mikefarah/yq#install)
```
Comment (timja) [inline README.md:9]:
```suggestion
- [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
- [yq](https://github.com/mikefarah/yq#install)
```

**Review by timja — APPROVED** [review]

**Pull request merged**